File: webapi/Models/HR/Contract.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;
using webapi.Models.System;

namespace webapi.Models.HR
{
    public record Contract
    {
        public int Id { get; set; }
        public Nullable<decimal> WorkingWage { get; set; }
        public Nullable<int> WorkTime { get; set; }
        public Nullable<Int16> AnnualLeave { get; set; }
        public string? AdditionalClause { get; set; }
        public DateOnly ConclusionDate { get; set; }
        public DateOnly ExecutionDate { get; set; }
        public Nullable<DateOnly> ContractTerm { get; set; }
        public Nullable<DateOnly> AdditionalAgreementDate { get; set; }
        public Nullable<DateOnly> TerminationDate { get; set; }
        public Nullable<DateOnly> ChangeDate { get; set; }


        public int ContractTypeId { get; set; }
        public SysContractType? ContractType { get; set; }


        public Nullable<int> SysPositionId { get; set; }
        public SysPosition? SysPosition { get; set; }


        public Nullable<int> SysIconomicActivityId { get; set; }
        public SysIconomicActivity? SysIconomicActivity { get; set; }


        public int DocumentTypeId { get; set; }
        public SysContractDocumentType? DocumentType { get; set; }


        public Nullable<int> TerminationTypeId { get; set; }
        public SysContractTerminationType? TerminationType { get; set; }


        public int SysAdministrativeTerritoryId { get; set; }
        public SysAdministrativeTerritory? SysAdministrativeTerritory { get; set; }


        [ForeignKey(nameof(ContractId))]
        public Nullable<int> ContractId { get; set; }
        [ForeignKey(nameof(ContractId))]
        public Contract? ParentContract { get; set; }


        public required string CompanyEic { get; set; }
        public required Int16 CodeCorection { get; set; }
        public Nullable<bool> IsTerminate {  get; set; }
        [Column("article62_flag")]
        public required bool Article62Flag {  get; set; }
        public required bool IsAnnex { get; set; }
        public DateOnly CreationDate { get; set; }
    }





    public record ContractDTO
    {
        public Nullable<decimal> WorkingWage { get; set; }
        public Nullable<int> WorkTime { get; set; }
        public Nullable<Int16> AnnualLeave { get; set; }
        public string? AdditionalClause { get; set; }
        public DateOnly ConclusionDate { get; set; }
        public DateOnly ExecutionDate { get; set; }
        public Nullable<DateOnly> ContractTerm { get; set; }
        public Nullable<DateOnly> AdditionalAgreementDate { get; set; }
        public Nullable<DateOnly> TerminationDate { get; set; }
        public Nullable<DateOnly> ChangeDate { get; set; }
        public int ContractTypeId { get; set; }
        public Nullable<int> SysPositionId { get; set; }
        public Nullable<int> SysIconomicActivityId { get; set; }
        public Int16 DocumentTypeId { get; set; }
        public Nullable<int> TerminationTypeId { get; set; }
        public Nullable<int> SysAdministrativeTerritoryId { get; set; }
        public required string CompanyEic { get; set; }
        public Nullable<int> ContractId { get; set; }
        public Int16 CodeCorection { get; set; }
        public bool Article62Flag { get; set; }
        public bool IsAnnex { get; set; }
    }
}

File: reactapp/src/models/HR/Contract.ts
import { Optional } from "@app-types/Optional";

export interface Contract {
    id: int,
    workingWage: Optional<string>;
    workTime: Optional<int>;
    annualLeave: Optional<int>;
    conclusionDate: Date;
    executionDate: Date;
    contractTerm: Optional<Date>;
    additionalAgreementDate: Optional<Date>;
    terminationDate: Optional<Date>;
    changeDate: Optional<Date>;
    contractTypeId: int,
    sysPositionId:Optional<int>,
    sysIconomicActivityId: Optional<int>,
    documentTypeId: int,
    terminationTypeId: Optional<int>,
    sysAdministrativeTerritoryId: Optional<int>,
    codeCorection: int,
    isTerminate: boolean,
    article62Flag: boolean,
}

export const defaultContractUpdateDTO: Contract = {
    id: 0,
    workingWage: "",
    workTime: 0,
    annualLeave: 0,
    conclusionDate: new Date(),
    executionDate: new Date(),
    contractTerm: null,
    additionalAgreementDate: null,
    terminationDate: null,
    changeDate: null,
    contractTypeId: 0,
    sysPositionId:0,
    sysIconomicActivityId: 0,
    documentTypeId: 0,
    terminationTypeId: 0,
    sysAdministrativeTerritoryId: 0,
    codeCorection: 0,
    isTerminate: false,
    article62Flag: false,
}

export interface ContractInsertDTO {
    workingWage: Optional<string>;
    workTime: Optional<int>;
    annualLeave: Optional<int>;
    additionalClause: Optional<string>,
    conclusionDate: Date;
    executionDate: Date;
    contractTerm: Optional<Date>;
    additionalAgreementDate: Optional<Date>;
    contractTypeId: int;
    sysPositionId: Optional<int>;
    sysIconomicActivityId: Optional<int>;
    documentTypeId: int;
    companyEic: string,
    contractId: Optional<int>,
    sysAdministrativeTerritoryId: Optional<int>;
    codeCorection: int;
    isAnnex: boolean,
    article62Flag: boolean,
}

export const defaultContractInsert: ContractInsertDTO = {
    workingWage: "",
    workTime: null,
    annualLeave: null,
    additionalClause: null,
    conclusionDate: new Date(),
    executionDate: new Date(),
    contractTerm: null,
    additionalAgreementDate: null,
    contractTypeId: 0,
    sysPositionId: null,
    sysIconomicActivityId: null,
    documentTypeId: 0,
    companyEic: "",
    contractId: null,
    sysAdministrativeTerritoryId: null,
    codeCorection: 0,
    isAnnex: false,
    article62Flag: false,
}